fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -523,9 +529,13 @@ void FixedLagSmoother::transactionCallback(
       else
       {
         // And purge out old transactions to limit the pending size while waiting for an ignition sensor
-        auto purge_time = ros::Time(0, 0);
+        auto purge_time = rclcpp::Time(0, 0, RCL_ROS_TIME);  // NOTE(CH3): Uninitialized
         auto last_pending_time = pending_transactions_.front().stamp();
-        if (ros::Time(0, 0) + params_.transaction_timeout < last_pending_time)  // ros::Time doesn't allow negatives
+
+        // rclcpp::Time doesn't allow negatives
+        // NOTE(CH3): In this case we're okay with uninitialized time since it's just used
+        //            for comparison
+        if (rclcpp::Time(0, 0, RCL_ROS_TIME) + params_.transaction_timeout < last_pending_time)
```

Comment:
Not sure what adding `0` time does here. Is it to convert a duration to a time for the comparison? If so, `rclcpp::Time(params_.transation_timeout.nanoseconds(0, last_pending_time.get_clock_type())` might be a better fit because it makes sure the clock types are comparable.

Reply:
Fair point, I'll update the one on top also

Reply:
https://github.com/locusrobotics/fuse/pull/283/commits/c16b02f15441790cdeea5bcd023940801ca89a01